src/main/java/com/emt/service/MatchService.java

```diff
@@ -1,41 +1,60 @@
 package com.emt.service;
 
+import com.emt.entity.Player;
 import com.emt.mapper.MatchMapper;
 import com.emt.model.exception.IdenticalPlayersException;
 import com.emt.model.request.CreateMatchRequest;
 import com.emt.model.response.MatchResponse;
 import com.emt.repository.MatchRepository;
+
 import java.util.List;
-import java.util.Optional;
+import java.util.stream.Stream;
+
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class MatchService {
 
-  private final MatchRepository matchRepository;
-  private final MatchMapper matchMapper;
+    private final MatchRepository matchRepository;
+
+    private final MatchMapper matchMapper;
+
+    private final PlayerService playerService;
+
+    private static final int CONSTANT_K = 30;
+
+    public List<MatchResponse> getAllMatches() {
+        return matchRepository.findAll().stream().map(matchMapper::mapToResponse).toList();
+    }
+
+    public MatchResponse createMatch(CreateMatchRequest request) {
+        if (request.winnerId().equals(request.loserId())) {
+            throw new IdenticalPlayersException("A match cannot be created with identical players.");
+        }
+
+        Player winner = playerService.getPlayerById(request.winnerId());
+        Player loser = playerService.getPlayerById(request.loserId());
 
-  private final PlayerService playerService;
+        updateEloRatings(winner, loser);
 
-  public List<MatchResponse> getAllMatches() {
-    return matchRepository.findAll().stream().map(matchMapper::mapToResponse).toList();
-  }
+        return Stream.of(matchMapper.mapToEntity(winner, loser))
```

Comment:
Why is it a `stream.of`? It's always a single element, and non-null, use `Optional.of`